relay-general/src/protocol/logentry.rs

```diff
@@ -10,11 +10,11 @@ use crate::types::{Annotated, Error, FromValue, Meta, Object, Value};
 pub struct LogEntry {
     /// The log message with parameter placeholders.
     #[metastructure(max_chars = "message")]
-    pub message: Annotated<String>,
+    pub message: Annotated<Message>,
 
     /// The formatted message
     #[metastructure(max_chars = "message", pii = "true")]
-    pub formatted: Annotated<String>,
+    pub formatted: Annotated<Message>,
```

Comment:
Is there any way we could annotate the value type on the field rather than creating a wrapper type just for that? We're creating a lot of newtypes for such simple things that could be on the field. Serde could serve as inspiration here (since we're reinventing the wheel already).

Reply:
I don't think we could introduce this *instead* of the type-level attribute. 90% of the time it's necessary to annotate it on the actual type.